x/service/types/genesis.go

```diff
@@ -29,6 +30,17 @@ func (gs GenesisState) Validate() error {
 		}
 		serviceNameMap[serviceName] = struct{}{}
 	}
+
+	// Check for duplicated index in relayMiningDifficulty
+	relayMiningDifficultyIndexMap := make(map[string]struct{})
+
+	for _, elem := range gs.RelayMiningDifficultyList {
+		index := string(RelayMiningDifficultyKey(elem.ServiceId))
+		if _, ok := relayMiningDifficultyIndexMap[index]; ok {
+			return ErrServiceDuplicateIndex.Wrapf("duplicated index for relayMiningDifficulty: %s", index)
+		}
+		relayMiningDifficultyIndexMap[index] = struct{}{}
+	}
```

Comment:
Wdyt about extracting these to methods (e.g. `#validateServiceList()` & `#validateRelayMiningDifficultyList()`)?